Dropout.lua

```diff
@@ -0,0 +1,70 @@
+local Dropout, Parent = nn.Dropout, nn.Module
```

Comment:
@nicholas-leonard I added `lazy` option to only resample after backward is called. This mechanism is used by Bayesian GRUs to use the same dropout mask for each sequence, not for each word. See [A Theoretically Grounded Application of Dropout in Recurrent Neural Networks by Gal (2015)](http://arxiv.org/abs/1512.05287).